libs/error-stack/src/report.rs

```diff
@@ -471,6 +554,24 @@ impl<C> Report<C> {
     }
 }
 
+impl<E: Into<Report<E>>> From<E> for Report<dyn Error + Send + Sync + 'static> {
```

Comment:
`dyn Error` is only a marker type, it does not really serve any purpose. In theory, it could be anything, such as `()`. `dyn Error` is descriptive and `Report<C>` always implements `Send + Sync` as it only can be constructed with such a type. It should be enough to just use
```suggestion
impl<E: Into<Report<E>>> From<E> for Report<dyn Error> {
```
This also aligns better with not setting a default for `C` as it’s less to type.

Reply:
> In theory, it could be anything, such as ().
It couldn't be () as that's `Sized` and you'd lose the cool ability to specialise methods that don't apply to `dyn Error`.

You're right on the `Send + Sync + 'static`, cool! Updated.